Added Battery Widget

vkylamba opened the pull request:
Updated.
![image](https://cloud.githubusercontent.com/assets/13446335/24071372/18543e1a-0bf6-11e7-9b18-611fc9109ed3.png)

Paul-Reed commented:
Thanks for the PR, however it doesn't seem to work here as intended. For example; The widget is configured like this;

![settings](https://cloud.githubusercontent.com/assets/973580/24063716/b6559210-0b58-11e7-93d2-0e3e75cbf59b.PNG)

and the feed input is 151, I get;

![battery](https://cloud.githubusercontent.com/assets/973580/24063727/c0faccb2-0b58-11e7-89c5-73744a400e59.PNG)

Also, I see that the text size does not scale with the size of the widget, or is user definable - font, label colour (see the bar widget - also in the screenshot above)
It would be good to have some formatting consistency across the widgets.

Paul

vkylamba commented:
@Paul-Reed Thanks for the review. I have updated the PR. Please have a look.

Paul-Reed commented:
That's better, it displays the animation OK now.

1. Do we need a widget title? we could of course have several battery widgets displayed on the same page (such as banks of batteries) but with no title, it may not be easy to identify which widget relates to which battery.
2. I think variable rounding of the displayed value would be useful, such as https://github.com/emoncms/dashboard/blob/master/widget/bar/bar_render.js#L252-L258 to enable different precision depending upon the value size. If you were monitoring 999V is there any need to display precision to 0.1V. Similarly for a 1V battery it would be useful to display the value to greater precision.
3. The text value does not appear to be centre aligned to the widget - see screenshot below using Arial Back typeface.
4. Also, for consistency, could the font size be increased to match that of the bar widget? In the screenshot, both widgets are the same size, yet the battery text size is smaller. (so that we have some dashboard consistency)

Paul

![align](https://cloud.githubusercontent.com/assets/973580/24074085/008b6ade-0bfa-11e7-887a-f390e2115797.PNG)

vkylamba commented:
1. Added widget title. 
2. Corrected the rounding off. 
3. Text centre aligned.
4. Fixed font size.
![image](https://cloud.githubusercontent.com/assets/13446335/24080096/ba5eee04-0cbd-11e7-9364-21c1e0cbde4a.png)

Paul-Reed commented on widget/battery/battery_render.js:
```diff
@@ -0,0 +1,183 @@
+    [5, "Helvetica"],
+    [4, "Comic Sans MS"],
+    [3, "Courier New"],
+    [2, "arial"],
```
arial is already listed on line 40

Paul-Reed commented on widget/battery/battery_render.js:
```diff
@@ -0,0 +1,183 @@
+      if (font == 5){fontname = "Helvetica"}
+      if (font == 6){fontname = "sans-serif"}
+      if (font == 7){fontname = "Arial Narrow"}
+      if (font == 8){fontname = "Arial"}
```
Arial is already listed on line 95

Paul-Reed commented:
Thanks for your patience Vikas, the battery widget looks good and matches the existing widgets nicely.

![align](https://cloud.githubusercontent.com/assets/973580/24080475/dd86d82c-0c97-11e7-9cd4-80a9e979dd6c.PNG)

Paul

Paul-Reed commented:
Vikas, I've made a forum entry about the widget https://community.openenergymonitor.org/t/battery-widget-now-added-to-dashboard/3473
I can't find your ID in the forum, are you a forum member? It would be good to introduce yourself in the thread.

Thanks

Paul

vkylamba commented:
Thank you Paul for merging the PR. I wasn't a member till now. Just signed up.

glynhudson commented:
Nice work, looks good. Thanks guys!

- sent from my mobile device

On 19 Mar 2017 13:10, "Vikas" <notifications@github.com> wrote:

> Thank you Paul for merging the PR. I wasn't a member till now. Just signed
> up.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/emoncms/dashboard/pull/76#issuecomment-287612254>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAuUPMjnc0eEdanh2CEKA2QORs1ddhwRks5rnRs7gaJpZM4MgbDX>
> .
>
